nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+      preStart = ''
+        umask 0077
+        mkdir -p ${dataDir}/etc
+        mkdir -p ${dataDir}/spool
+        mkdir -p ${dataDir}/list_data
+        mkdir -p ${dataDir}/arc
+        mkdir -p ${dataDir}/bounce
+
+        cp -f ${mainConfig} ${dataDir}/etc/sympa.conf
+        DBPASS="$(head -n1 ${cfg.database.passwordFile})"
+        if [ -n "$DBPASS" ]; then
+            sed -e "s,#dbpass#,$DBPASS,g" \
+                -i ${dataDir}/etc/sympa.conf
+        else
+            sed -e "/db_passwd.*#dbpass#/d" \
+                -i ${dataDir}/etc/sympa.conf
+        fi
+
+        ${concatStringsSep "\n" (flip mapAttrsToList cfg.domains (fqdn: domain:
+        ''
+          mkdir -p -m 750 ${dataDir}/etc/${fqdn}
+          cp -f ${robotConfig fqdn domain} ${dataDir}/etc/${fqdn}/robot.conf
+          mkdir -p -m 750 ${dataDir}/list_data/${fqdn}
+        ''))}
+
+        cp -f ${virtual} ${dataDir}/virtual.sympa
+        cp -f ${transport} ${dataDir}/transport.sympa
+        cp -f ${listAliases} ${dataDir}/etc/list_aliases.tt2
+
+        touch ${dataDir}/sympa_transport
+
+        ${optionalString (cfg.mta.type == "postfix") ''
+          ${pkgs.postfix}/bin/postmap hash:${dataDir}/virtual.sympa
+          ${pkgs.postfix}/bin/postmap hash:${dataDir}/transport.sympa
+        ''}
+        ${pkg}/bin/sympa_newaliases.pl
+
+
+        cp -af ${pkg}/static_content ${dataDir}/
+        # Yes, wwsympa needs write access to static_content..
+        chmod -R u+w,a+X ${dataDir}/static_content/
+        # Make static_content accessible to nginx
+        chmod o+X ${dataDir}
+
+        ${pkg}/bin/sympa.pl --health_check
+      '';
+    };
+    systemd.services.sympa-archive = {
+      description = "Sympa mailing list manager (archiving)";
+      bindsTo = [ "sympa.service" ];
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
+      serviceConfig = sympaServiceConfig "archived";
+    };
+    systemd.services.sympa-bounce = {
+      description = "Sympa mailing list manager (bounce processing)";
+      bindsTo = [ "sympa.service" ];
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
+      serviceConfig = sympaServiceConfig "bounced";
+    };
+    systemd.services.sympa-bulk = {
+      description = "Sympa mailing list manager (message distribution)";
+      bindsTo = [ "sympa.service" ];
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
+      serviceConfig = sympaServiceConfig "bulk";
+    };
+    systemd.services.sympa-task = {
+      description = "Sympa mailing list manager (task management)";
+      bindsTo = [ "sympa.service" ];
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
+      serviceConfig = sympaServiceConfig "task_manager";
+    };
+
+    systemd.services.wwsympa = mkIf usingNginx {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "sympa.service" ];
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
+      serviceConfig = {
+        Type = "forking";
```

Comment:
Consider making this run in the foreground as well

Reply:
I've changed it to run in foreground - the downside is that it's not possible to have more than one worker process now, which is OK for my deployment but might not be sufficient for others.

Reply:
What you have described qualifies as a good reason to run as `forking`. I'm surprised to hear that it cannot have more than one worker process this way, though.

Reply:
Yes, the documentation states that:
```
       -n      No forking should take place (for daemontools)
```
There are [ways around it](https://nileshgr.com/2016/07/09/systemd-fastcgi-multiple-processes) but the complexity compared to `forking` is probably not worth it. Will bring `forking` back.